### ISSUE TITLE
When performing a query selecting on all primary keys, any match clause will be ignored.

### ISSUE THREAD

**Issue by ghotikun** [crate/crate#4555]
**CrateDB version**: 0.56.3

**JVM version**: 25.92-b14

**OS version / environment description**:

**Problem description**:

When selecting on all available primary keys, no other keys, and a match clause, crate will return all responses that match the primary keys---even if they don't match the match clause.

**Steps to reproduce**:

```
CREATE TABLE primary_key_bug (
  primary1 STRING,
  primary2 STRING,
  field STRING,
  content STRING INDEX USING plain,
  INDEX english USING fulltext(content) with (analyzer = 'english'),
  PRIMARY KEY (primary1, primary2)
)
```

`insert into primary_key_bug (primary1, primary2, field, content) values ('A', '1', 'one', 'apple');
`

Selecting on all primary keys and a match
`select * from primary_key_bug where primary1='A' AND primary2='1' AND match(english, 'pear');
`
incorrectly returns: ('apple', 'one', 'A', '1')

Selecting on some primary keys and a match
`select * from primary_key_bug where primary1='A' AND match(english, 'pear');
`
correctly returns nothing

Selecting on all primary keys, a regular field, and a match
`select * from primary_key_bug where primary1='A' AND primary2='1' AND field='one' AND match(english, 'pear');
`
correctly returns nothing

Selecting on all primary keys and a regular field
`select * from primary_key_bug where primary1='A' AND primary2='1' AND field='two';
`
correctly returns nothing

**Use case**:

We want to limit a match clause to a subset of rows. We do this by restricting the query to all rows that belong to a subset of a particular primary key, _recordid_. For example we include `recordid in ('1', '2', '3')` in the where clause to restrict the match clause to rows with these recordids. Unfortunately, this always returns all rows in the subset of recordids, even if they don't match the match clause. We can get around this by including a superfluous field in the query (example: `recordid in ('1', '2', '3') AND samplevalue!=-1`). But this is clearly a bug.

**Feature description**:

**Comment by mfussenegger** [comment]
uh good catch. Thanks for reporting.

**Comment by mfussenegger** [comment]
Fixed with this https://github.com/crate/crate/pull/4561

The fix will be included in the next 0.57.X / 1.0.X releases.

**Comment by ghotikun** [comment]
Thank you!
